### ISSUE TITLE
Channels Tags in sessionInfo

### ISSUE THREAD

**Issue by dlevenstein** [buzsakilab/buzcode#183]
add option to sessionInfoGUI to add arbitrary "tags" for channels.  
For example: ThetaChannel, CTXRepresentativeChannel, etc etc. 

Can even have drop-down menu for commonly used tags.

**Comment by brendonw1** [comment]
I’m not sure how to think about session info.  Are we turning it into a legitimate metadata structure?   Right now it’s a limited metadata structure - is the idea to really change that?  Is if the right place to put this stuff.  It doesn’t quite strike me as right but maybe I just need to reconceptualize sessioninfo.

**Comment by dlevenstein** [comment]
Maybe think of it as metadata for analysis.

**Comment by brendonw1** [comment]
I’m basically fine w that. But I will say it’s like a weird slippery slope
in my mind. Like all metadata is useful for analysis - depending on the
depth of the analysis. So where to draw the line will be unclear. But it’s
prob fine for now. I think what it really is is that I want a more complete
metadata solution. I hate not gathering the data with a fully understood
metadata framework ahead of time.

**Comment by brendonw1** [comment]
For, say ripple channel, I'm thinking of making a field in a field inside sessionInfo, like this: sessionInfo.LFPAnalysisChannels.RippleChannel.  OK?

**Comment by DavidTingley** [comment]
Although, wouldn't we want this metadata stored in the ripples.event file? as it pertains to channels used for that.

I'm wondering how we can keep these things synchronized..

**Comment by dlevenstein** [comment]
Yes definitely.  Channels used for detection should always we stored in the events.mat file.

Especially since you might have (for some unknown but maybe interesting physiological reason) two different ripple events on two different channels. (Maybe from two different regions? ;))

**Comment by brendonw1** [comment]
I’d like to have them somewhere BEFORE ripple detection so ripple detection
can look for it somewhere and use it.
SessionInfo isn’t my favorite but it’s ok.

Dan what about “TaggedChannels”?

**Comment by DavidTingley** [comment]
@brendonw1, I agree having detectors inherit from sessionInfo is a great idea.  We're getting closer and closer to OOP buzcode...

**Comment by dlevenstein** [comment]
I'd prefer it be more general. A box for any channel tags you may have.  

for example, badchannels should move to sessionInfo.ChannelTags.badchannels

**Comment by dlevenstein** [comment]
I dunno, I thought the term tag was pretty standard/general - this is where you store any tags (labels) you want to associate with specific channels. Maybe that didn't get across or isn't how you're thinking about it?  Seems like a good general feature to have. I don't really have another term for the same thing. ChannelLabels sounds clunky to me but could work I guess.

https://en.wikipedia.org/wiki/Tag_(metadata)

**Comment by brendonw1** [comment]
I worry that it seems like a temporary term.  But that might be because I’m old.  I also think “coding” seems that way too :)

Basically I’m fine with it if others are fine ... @davidtingley - does “tag” seem like a solid term to use?   If you’re fine I’m fine.

**Comment by brendonw1** [comment]
The resounding silence on this issue convinces me that no one else cares and we should just do "channel tags"

**Comment by dlevenstein** [comment]
🦗🦗🦗

**Comment by brendonw1** [comment]
???

**Comment by dlevenstein** [comment]
^^^crickets

**Comment by brendonw1** [comment]
Ah ;-P
So how do we move forward?

**Comment by dlevenstein** [comment]
Oh, the crickets didn't look very crickety on here.... I guess github doesn't have good emoji integration.

Here's a possible moving forward plan of attack:
1) Update SessioninfoGUI to have a ChannelTags section
2) make a dropdown list of common channel tags, and an option for the user to add their own. users will select or enter a channel tag (i.e. RippleChannel) and list the channels they want associated with that tag.  Will be saved as sessionInfo.ChannelTags.RippleChannel 

3) move badchannels to a common ChannelTag
4) and to avoid issues with backwards compatibility of badchannels, make bz_getSessionInfo move sessionInfo.badchannels to sessionInfo.ChannelTags.badchannels if it finds them there when loading a sessionInfo file
5) Update existing functions that reference sesionInfo.badchannels to reference sessionInfo.ChannelTags.badchannels

**Comment by brendonw1** [comment]
Sounds perfect!

**Comment by dlevenstein** [comment]
Step 1-3 is on my short time scale to-do list. Let me know if you dive into it first.

**Comment by dlevenstein** [comment]
Update on this: I've added the function bz_TagChannel(basePath,channums,tag) (see #294)

The function is an easy way for the user to add "tags" to specific channels in the sessionInfo. It saves any tags in: sessionInfo.channelTags.(tagName)
For example, it's now being used in PickSWTHChannel to save sessionInfo.channelTags.NREMDetectionChan and sessionInfo.channelTags.ThetaChan

I haven't yet integrated into SessionInfoGUI (steps 1-2 above), but would like to do so, and move badchannels and all references to badchannels in the repo over to this form. (i.e. implement 3-5 above). Does anyone have a problem with this, or thoughts to improve documentation/implementation?
